Library/OcBootManagementLib/BootArguments.c

```diff
@@ -360,3 +374,226 @@ OcCheckArgumentFromEnv (
 
   return HasArgument;
 }
+
+EFI_STATUS
+OcParseLoadOptions (
+  IN     CONST EFI_LOADED_IMAGE_PROTOCOL  *LoadedImage,
+     OUT       OC_DYNAMIC_ARRAY           **ParsedLoadOptions
+  )
+{
+  CHAR16                      *Pos;
+  LOAD_OPTIONS_PARSE_STATE    State;
+  BOOLEAN                     Retake;
+  CHAR16                      c;
+  CHAR16                      *Name;
+  CHAR16                      *Value;
+  OC_LOAD_OPTION              *Option;
+
+  ASSERT (LoadedImage != NULL);
+  ASSERT (ParsedLoadOptions != NULL);
+  *ParsedLoadOptions = NULL;
+
+  if (LoadedImage->LoadOptionsSize > MAX_LOAD_OPTIONS_SIZE) {
```

Comment:
This lacks evenness checking. If LoadOptionsSize is not divisible by sizeof(CHAR16), it is not valid.